Sources/NIO/ByteBuffer-views.swift

```diff
@@ -21,43 +21,61 @@ public struct ByteBufferView: RandomAccessCollection {
     public typealias Index = Int
     public typealias SubSequence = ByteBufferView
 
-    fileprivate let buffer: ByteBuffer
-    fileprivate let range: Range<Index>
+    /* private but usableFromInline */ @usableFromInline var _buffer: ByteBuffer
+    /* private but usableFromInline */ @usableFromInline var _range: Range<Index>
 
     internal init(buffer: ByteBuffer, range: Range<Index>) {
         precondition(range.lowerBound >= 0 && range.upperBound <= buffer.capacity)
-        self.buffer = buffer
-        self.range = range
+        self._buffer = buffer
+        self._range = range
+    }
+
+    /// Creates a `ByteBufferView` from the readable bytes of the given `buffer`.
+    public init(_ buffer: ByteBuffer) {
+        self = ByteBufferView(buffer: buffer, range: buffer.readerIndex ..< buffer.readerIndex + buffer.readableBytes)
```

Comment:
Nit: `buffer.readerIndex + buffer.readableBytes` is `buffer.writerIndex`, so we can just use that directly.